File: TradingAppLibrary/Models/Candle.cs
namespace TradingAppLibrary.Models;

public class Candle
{
    public long Id { get; set; }
    public int SymbolId { get; set; }
    public Symbol Symbol { get; set; } = null!;
    public string Interval { get; set; } = "1m";
    public DateTime OpenTime { get; set; }
    public decimal Open { get; set; }
    public decimal High { get; set; }
    public decimal Low { get; set; }
    public decimal Close { get; set; }
    public decimal Volume { get; set; }
}

File: tradingappfrontend/src/Types/candle.ts
export interface CandleData {
    time: number;
    open: number;
    high: number;
    low: number;
    close: number;
    volume?: number;
    symbolId?: number;
    symbol?: string;
    exchange?: string;
    interval?: string;
    isClosed?: boolean;
}
export type ChartType = 'candles' | 'bars' | 'line' | 'area' | 'heikin-ashi';
